src/functions/api.js

```diff
@@ -24,12 +59,24 @@ API.prototype.request = function request(method, data) {
 };
 
 const methods = ['getMe', 'sendMessage', 'forwardMessage', 'sendPhoto',
-'sendAudio', 'sendDocument', 'sendSticker', 'sendVideo',
-'sendLocation', 'sendChatAction', 'getUserProfilePhotos',
-'getUpdates', 'setWebhook', 'deleteMessage'];
+  'sendAudio', 'sendDocument', 'sendSticker', 'sendVideo',
+  'sendLocation', 'sendChatAction', 'getUserProfilePhotos',
+  'getUpdates', 'setWebhook', 'deleteMessage'];
 
 methods.forEach(method => {
-  API.prototype[method] = function(data) { //eslint-disable-line
-    return this.request(method, data);
+  API.prototype[method] = function (data) { //eslint-disable-line
+    // implementation taken from https://github.com/yagop/node-telegram-bot-api/issues/192#issuecomment-249488807
+    let resolve;
+    let reject;
+
+    const promise = new Promise((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
+    });
+
+    this._queue.push({ method, data, resolve, reject });
```

Comment:
Why not wrap this function inside the promise, so you don't have to hold references to `resolve` and `reject`.

```javascript

return new Promise((resolve, reject) => {
  this._queue.push({ method, data, resolve, reject });
  process.nextTick(this._runQueue.bind(this));
});
```

Reply:
You are right, after re-running same test file that sends out multiple messages got same result without any issues. Appears there's no negative impact reducing this code excerpt you suggested :+1: 
Added your suggestions in a new commit